deepchecks/utils/abstracts/data_duplicates.py

```diff
@@ -0,0 +1,50 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2023 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""module contains base logic for the data duplicates checks."""
+import typing as t
+import abc
+
+from typing_extensions import Self
+
+from deepchecks.core import ConditionCategory, ConditionResult
+from deepchecks.utils.strings import format_percent
+
+__all__ = ['DataDuplicatesAbstract']
+
+
+class DataDuplicatesAbstract(abc.ABC):
```

Comment:
Optional - See if you can add more stuff in the abstract class:
I think the nlp and tabular check display + value can be the same after you get the list of duplicated indexes

Reply:
I suggest creating a separate issue for this and all other checks

The problem is that the checks work similarly but not quite similarly/
It is not obvious how to extract/separate their common logic into a generic function that will be useful, clean, and easy to use in both checks

so, currently, I would prefer how it is

Reply:
So Yuri can you open that issue please?

Reply:
created
[DEE-520](https://linear.app/deepchecks/issue/DEE-520/common-logic-abstract-for-tabular-and-nlp-integrity-checks)